tests/test_intercept.js

```diff
@@ -196,6 +196,23 @@ test("reply should throw on error on the callback", function(t) {
   req.end();
 });
 
+test('reply should not cause an error on header conflict', function (t) {
+  const scope = nock('http://www.website.com/')
+    .defaultReplyHeaders({
+      'content-type': 'application/json'
+    });
+
+  t.doesNotThrow(() => {
+    scope
+      .post('/search')
+      .reply(200, '<html></html>', {
+        'Content-Type': 'application/xml'
+      });
+  });
+
+  t.end()
```

Comment:
@dcstone09 Before ending the test, could we actually make a request to `http://www.website.com/search`?IMO that would be a more complete assertion of the expected behavior.